ospf6d/ospf6_flood.c

```diff
@@ -842,18 +842,6 @@ void ospf6_receive_lsa(struct ospf6_neighbor *from,
 				zlog_debug("Received is duplicated LSA");
 			SET_FLAG(new->flag, OSPF6_LSA_DUPLICATE);
 		}
-		if (old->header->adv_router
-			    == from->ospf6_if->area->ospf6->router_id
-		    && OSPF6_LSA_IS_MAXAGE(new)) {
-			ospf6_acknowledge_lsa(new, ismore_recent, from);
-			ospf6_lsa_delete(new);
-			if (is_debug)
-				zlog_debug(
-					"%s: Received is self orig MAXAGE LSA %s, discard (ismore_recent %d)",
-					__PRETTY_FUNCTION__, old->name,
-					ismore_recent);
-			return;
-		}
```

Comment:
Just to verify -- it looks like the way the code is today, it simply discards a received duplicate of an LSA where the local copy is not MAXAGE but the received copy is. What this change does is to allow the next `if` to be taken (assuming `ismore_recent <0,` which I think it should be) so a new LSA is generated. RFC5340 just refers to 2328 here, rather than addressing this directly, and 2328 doesn't address the issue of receiving a self-originated LSA with MAXAGE directly. So the assumption is that MAXAGE is always "newer" than any existing age, and should trigger the creation of a new LSA (?).

Logically, this seems like the right thing to do--if someone has sent us a copy of our own LSA with MAXAGE set, and our local copy has some other age, then someone out there is flooding out an LSA of ours that will remove it from everyone else's LSDB. In this case, we should be forcing a refresh so everyone else has the correct information, rather than just discarding this LSA.

@dslicenc -- this seems right to me, but can you take a look to make certain I'm reading this all correctly?

Reply:
That is correct. RFC 2328 does not talk about receiving self originated LSA with MAX-AGE. In this case if the we have local router has restarted and hence the sequence number has reset too. If sequence number had not reset then we would not be here. The order in which we check for new LSA is sequence number, checksum and age. With discarding it we have issue where ospf neighbour will not even request for this LSA. There are two ways to solve this problem. First we can accept the LSA and install it in local DB, so that when we next refresh we will send latest LSA. Second way is to set the sequence number to received LSA sequence + 1 and refresh new LSA. Thoughts?

Reply:
Has Chirag taken a look at this change since it's directly reverting part of a fix of his?  It would be valuable to understand what misbehavior he was addressing with this code change in order to know what is now being put back (and may need to be dealt with in a different way.)

Reply:
@chiragshah6 can you take a look at this?

Reply:
I think the condition where the check was added the self originated MAX_AGE LSA come back with recency  was `-1` post frr restart which leads to not refreshing the local database copy. 

During the course of OSPFv3 `hardning` there was a instance where MAX_AGE LSA removed the recent copy but I could not find the bug-id or a specific test which leads to the condition. 

AS @riw777 pointed OSPFv3 RFC does not clarify the specific condition of MAX_AGE LSA processing. 

I have run the cumulus specific ospfv3 tests with this change and the results do pass. 

Also had a slack discussion with @Spantik  who has checked vendor implementation and the behavior is consistent.